fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/StringStatementRenderer.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query.expression;
+
+import static com.ibm.fhir.database.utils.query.SqlConstants.FROM;
+import static com.ibm.fhir.database.utils.query.SqlConstants.SELECT;
+import static com.ibm.fhir.database.utils.query.SqlConstants.SPACE;
+import static com.ibm.fhir.database.utils.query.SqlConstants.WHERE;
+
+import java.util.List;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.query.FromClause;
+import com.ibm.fhir.database.utils.query.FromItem;
+import com.ibm.fhir.database.utils.query.GroupByClause;
+import com.ibm.fhir.database.utils.query.HavingClause;
+import com.ibm.fhir.database.utils.query.OrderByClause;
+import com.ibm.fhir.database.utils.query.PaginationClause;
+import com.ibm.fhir.database.utils.query.SelectList;
+import com.ibm.fhir.database.utils.query.WhereClause;
+import com.ibm.fhir.database.utils.query.node.BindMarkerNode;
+import com.ibm.fhir.database.utils.query.node.ExpNode;
+
+/**
+ * Helps to render a select statement as a string
+ */
+public class StringStatementRenderer implements StatementRenderer<String> {
+
+    private static final String NEWLINE = System.lineSeparator();
```

Comment:
I've seen this in a couple places - move to a constants class?

Reply:
I'm deferring work on constants until we do the whole-system search. There will be a lot of consolidation and removal of unused values then.